### PR TITLE
docs: Add instructions for contributing with gitpod

### PR DESCRIPTION
**Pull request by svenefftinge**
Documentation on how to contribute to Gatsby using Gitpod.

Please advice where and how to put this information. 🙏

### CODE REVIEW

**Comment by svenefftinge** [comment]
@pieh 
I also wanted to point out that someone needs to install the [GitHub app](https://github.com/marketplace/gitpod-io) to have the pre-builds for all branches running. 
It reduces the boot time by ~5mins. Let me know if you have questions.

File: docs/contributing/using-an-online-dev-environment.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+To start working on an issue, you can prefix the issue URL with `gitpod.io/#`.
+
+> Example: https://gitpod.io/#https://github.com/gatsbyjs/gatsby/issues/1199
+
+This will open a fresh environment with a local branch named after the issue.
+You can now code, test, commit, push and create a PR from within the workspace.
+
+## Code Reviews
+
+Some code changes need a deeper review than what is possible on GitHub. Prefixing a PR with `gitpod.io/#` will open that branch in code review mode.
```
Comment (pieh) [inline docs/contributing/using-an-online-dev-environment.md]: Could we use similar example as in "Working on an Issue" section

File: docs/contributing/using-an-online-dev-environment.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+---
+title: Using an Online Dev Environment
```
Comment (pieh) [inline docs/contributing/using-an-online-dev-environment.md]: I would mention GitPod specifically here as this is GitPod specific documentation

**Comment by svenefftinge** [comment]
Hey @pieh, sorry for the bad experience. It was a bug that we have introduced only yesterday.
It works now and we are working on a reliable fix, which will go into production ASAP.
I'm currently on vacation and will update the PR early next week.

**Comment by svenefftinge** [comment]
@pieh I have force pushed your suggestion, but now the build fails. I think it is unrelated.

**Comment by sidharthachatterjee** [comment]
@svenefftinge Is this ready to go?

**Comment by svenefftinge** [comment]
@sidharthachatterjee Yes, from my side.

File: www/src/data/sidebars/contributing-links.yaml [hunk 3]
```diff
@@ -54,6 +54,8 @@
           link: /contributing/code-contributions/
         - title: Setting Up Your Local Dev Environment
           link: /contributing/setting-up-your-local-dev-environment/
+        - title: Using an Online Dev Environment
```
Comment (sidharthachatterjee) [inline www/src/data/sidebars/contributing-links.yaml]:
```suggestion

```

File: www/src/data/sidebars/contributing-links.yaml [hunk 4]
```diff
@@ -54,6 +54,8 @@
...
+          link: /contributing/using-an-online-dev-environment/
```
Comment (sidharthachatterjee) [inline www/src/data/sidebars/contributing-links.yaml]:
```suggestion

```

**Comment by svenefftinge** [comment]
@sidharthachatterjee thanks and yes, that's fine.

**Pull request merged**